fivekmrun_app_flutter/lib/timekeeping/chronometer_view.dart

```diff
@@ -0,0 +1,419 @@
+      setState(() {
+        _laps.add(_milliseconds);
+        _saveState();
+      });
+    }
+  }
+
+  Future<void> _deleteLap(int index) async {
+    final bool? shouldDelete = await showDialog<bool>(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: const Text('Delete Lap'),
+        content: const Text('Are you sure you want to delete this lap?'),
+        actions: [
+          TextButton(
+            onPressed: () => Navigator.of(context).pop(false),
+            child: const Text('Cancel'),
+          ),
+          TextButton(
+            onPressed: () => Navigator.of(context).pop(true),
+            child: const Text('Delete'),
+          ),
+        ],
+      ),
+    );
+
+    if (shouldDelete == true) {
+      setState(() {
+        _laps.removeAt(index);
+        _saveState();
+      });
+    }
+  }
+
+  String _formatTime(int milliseconds) {
+    int hundreds = (milliseconds / 10).floor() % 100;
+    int seconds = (milliseconds / 1000).floor() % 60;
+    int minutes = (milliseconds / 60000).floor() % 60;
+    int hours = (milliseconds / 3600000).floor();
+
+    String hoursStr = hours > 0 ? '${hours.toString().padLeft(2, '0')}:' : '';
+    return '$hoursStr${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}.${hundreds.toString().padLeft(2, '0')}';
+  }
+
+  Future<void> _exportToFile() async {
+    if (_laps.isEmpty) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(content: Text('No laps to export')),
+      );
+      return;
+    }
+
+    final now = DateTime.now();
+    final dayOfWeek = now.weekday == 1
+        ? 'Monday'
+        : now.weekday == 2
+            ? 'Tuesday'
+            : now.weekday == 3
+                ? 'Wednesday'
+                : now.weekday == 4
+                    ? 'Thursday'
+                    : now.weekday == 5
+                        ? 'Friday'
+                        : now.weekday == 6
+                            ? 'Saturday'
+                            : 'Sunday';
+
+    final dateStr =
+        '${now.year}-${now.month.toString().padLeft(2, '0')}-${now.day.toString().padLeft(2, '0')}';
+
+    String content = 'MCH:\t\t001\n\n';
+    content += 'Record Date:\t$dateStr\t$dayOfWeek\n\n';
+
+    int previousLapTime = 0;
+    for (int i = 0; i < _laps.length; i++) {
+      final lapNumber = i + 1;
+      final lapTime = _laps[i];
+      final splitTime = lapTime - previousLapTime;
+
+      final hours = (splitTime / 3600000).floor();
+      final minutes = (splitTime / 60000).floor();
+      final seconds = ((splitTime % 60000) / 1000).floor();
+      final milliseconds = (splitTime % 1000) ~/ 10;
+
+      final splitTimeStr =
+          '$hours:${minutes.toString().padLeft(2, '0')}\'${seconds.toString().padLeft(2, '0')}.${milliseconds.toString().padLeft(2, '0')}';
+
+      final totalHours = (lapTime / 3600000).floor();
+      final totalMinutes = (lapTime / 60000).floor();
```

Comment:
Looks like totalMinutes calculation is a bit off for times >1hour
Made some suggestions [here](https://github.com/5kmrun-bg/fivekmrun-app/pull/149)